### PR TITLE
add json logger

### PR DESCRIPTION
**Pull request by ericphanson**
I'm not totally sure this repo is the best place for this, but it probably should go somewhere. This adds a JSON logger based on LoggingFormats + LoggingExtras with some opinionated functionality for how to handle exceptions/stacktraces, and for logging by default some info like the timestamp + worker id.

Maybe the exception stuff would be welcome in LoggingFormats as the default behavior for `JSONFormat`? Or maybe it's too opinionated. cc @fredrikekre

### CODE REVIEW

File: src/logger.jl [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+"""
+    jsonable(obj::Any) = obj
+
+Add methods to `BeaconK8sUtilities.jsonable` in order to specify how `json_logger` should
+serialize types that themselves do not already have a `StructTypes.StructType` defined.
+"""
+jsonable(obj::Any) = obj
+
+function handle_log_exception(key, v)
```
Comment (fredrikekre) [inline src/logger.jl:14]: You mean this part for special-casing the `exception` keyword? That seems fine to upstream.

File: src/logger.jl [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    key == :exception || return v
+    if v isa Tuple && length(v) == 2
```
Comment (fredrikekre) [inline src/logger.jl:16]: I believe `@info "msg" exception=e` is also supported

File: src/logger.jl (hunk 1 above)
Comment (ericphanson) [reply src/logger.jl:14]: Yeah, that's what I meant. OK, sounds good, I will make a PR.

File: src/logger.jl (hunk 2 above)
Comment (ericphanson) [reply src/logger.jl:16]: I was doing `string(v)` for that, but I should do `sprint(showerror, e)` instead I think.

File: src/logger.jl [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+## Example
+
+For example, if we want to log `TimerOutput`s from TimerOutputs.jl, we can define the
+following.
+
+```julia
+using TimerOutputs
+
+BeaconK8sUtilities.jsonable(to::TimerOutput) = TimerOutputs.todict(to)
```
Comment (ericphanson) [inline src/logger.jl:20]: `TimerOutputs.todict` is only on master right now, so this will be a better example once we get a release

**Review by christopher-dG — APPROVED** [review]
Have been playing around with this in a pod, it looks great!

**Pull request merged**